prefect_gcp/cloud_storage.py

```diff
@@ -452,14 +451,195 @@ def example_cloud_storage_copy_blob_flow():
         dest_blob = source_blob
 
     source_blob_obj = source_bucket_obj.blob(source_blob)
-    partial_copy_blob = partial(
+    await run_sync_in_worker_thread(
         source_bucket_obj.copy_blob,
         blob=source_blob_obj,
         destination_bucket=dest_bucket_obj,
         new_name=dest_blob,
         timeout=timeout,
         **copy_kwargs,
     )
-    await to_thread.run_sync(partial_copy_blob)
 
     return dest_blob
+
+
+class GcsBucket(ReadableFileSystem, WritableFileSystem):
```

Comment:
These are the ABCs for deployment storage blocks now.
```suggestion
class GcsBucket(WritableFileSystem, WritableDeploymentStorage):
```